support-payment-api/src/main/scala/backend/AmazonPayBackend.scala

```diff
@@ -66,22 +73,35 @@ class AmazonPayBackend(
       )
     }
 
-    val response = for {
-      orderRef <- service.getOrderReference(amazonPayRequest.paymentData.orderReferenceId)
-      _ <- if (isNotSuspended(orderRef)) service.setOrderReference(amazonPayRequest.paymentData) else Right(orderRef)
-      _ <- service.confirmOrderReference(orderRef)
-      authRes <- service.authorize(orderRef, amazonPayRequest.paymentData)
-      _ <- handleDeclinedResponse(orderRef, authRes.getAuthorizationStatus)
-    } yield authRes
+    amazonPayEnabled.flatMap {
+      case true =>
+        val response =
+          for {
+            orderRef <- service.getOrderReference(amazonPayRequest.paymentData.orderReferenceId)
+            _ <-
+              if (isNotSuspended(orderRef)) service.setOrderReference(amazonPayRequest.paymentData) else Right(orderRef)
+            _ <- service.confirmOrderReference(orderRef)
+            authRes <- service.authorize(orderRef, amazonPayRequest.paymentData)
+            _ <- handleDeclinedResponse(orderRef, authRes.getAuthorizationStatus)
+          } yield authRes
+
+        handleResponse(response, amazonPayRequest, clientBrowserInfo)
+      case _ =>
+        handleResponse(
+          Either.left(AmazonPayApiError.fromString(amazonPayErrorText)),
+          amazonPayRequest,
+          clientBrowserInfo,
+        )
+
+    }
```

Comment:
I think breaking part of this out into a separate method makes it easier to read?

```suggestion
    def getAuthorizationDetails = for {
      orderRef <- service.getOrderReference(amazonPayRequest.paymentData.orderReferenceId)
      _ <-
        if (isNotSuspended(orderRef)) service.setOrderReference(amazonPayRequest.paymentData) else Right(orderRef)
      _ <- service.confirmOrderReference(orderRef)
      authRes <- service.authorize(orderRef, amazonPayRequest.paymentData)
      _ <- handleDeclinedResponse(orderRef, authRes.getAuthorizationStatus)
    } yield authRes

    amazonPayEnabled.flatMap {
      case true =>
        handleResponse(getAuthorizationDetails, amazonPayRequest, clientBrowserInfo)
      case false =>
        handleResponse(Left(AmazonPayApiError.fromString(amazonPayErrorText)), amazonPayRequest, clientBrowserInfo)
    }
```

Reply:
updated as per suggestion